File: src/dotnet/yaxgl/YaxglLibrary/Base/Control.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace de.yaxgl
{
    public abstract class Control : Component,Containable
    {
        
        
        
        public void clickEvent(object sender, System.EventArgs e)
        {
            owner.notifyEvent(this, new EventArgs(EventType.Click, this.ID));
        }

        public void gotFocusEvent(object sender, System.EventArgs e)
        {
            owner.notifyEvent(this,new EventArgs(EventType.GotFocus, this.ID));
        }

        public void lostFocusEvent(object sender, System.EventArgs e)
        {
            owner.notifyEvent(this, new EventArgs(EventType.LostFocus, this.ID));
        }
        
        public void selectionChangedEvent(object sender, System.EventArgs e)
        {
            owner.notifyEvent(this, new EventArgs(EventType.SelectionChanged, this.ID));
        }

        public void setEnabled(bool enabled)
        {
            this.control.Enabled = enabled;
        }

        public bool isEnabled()
        {
            return this.control.Enabled;
        }   
    }
}

File: trunk/src/python/src/de/yaxgl/Base/Control.py
from de.yaxgl.Base.Component import Component
from de.yaxgl.EventDispatcher.EventArgs import EventArgs
from de.yaxgl.EventDispatcher.Decorator.EventType import EventType



# abstract
class Control(Component):
    
    def clickEvent(self, sender, e):
        self.owner.notifyEvent(self, EventArgs(EventType.Click, self.ID))
        
    def gotFocusEvent(self, sender, e):
        self.owner.notifyEvent(self, EventArgs(EventType.GotFocus, self.ID))
        
    def lostFocusEvent(self, sender, e):
        self.owner.notifyEvent(self, EventArgs(EventType.LostFocus, self.ID))
        
    def selectionChangedEvent(self, sender, e):
        self.owner.notifyEvent(self, EventArgs(EventType.SelectionChanged, self.ID))
        
    def setEnabled(self, enabled):
        self.control.Enabled = enabled
        
    def isEnabled(self):
        return(self.control.Enabled) 
